src/components/Box/Box.tsx

```diff
@@ -244,7 +285,34 @@ export const Box: FC<BoxProps> = forwardRef((
   const isFlexBox = typeof display === 'string' && display.includes('flex');
   const flexDirectionClasses = isFlexBox ? generateResponsiveClasses('flex-direction', direction) : null;
 
+  const cssPropertyMap: {
+    [key: string]: {
+      classPrefix: string;
+      transformer: typeof generateResponsiveClasses | typeof cssShorthandToClasses;
+    };
+  } = {
+    color: { classPrefix: 'font-color', transformer: generateResponsiveClasses },
+    background: { classPrefix: 'background-color', transformer: generateResponsiveClasses },
+    borderColor: { classPrefix: 'border-color', transformer: generateResponsiveClasses },
+    borderWidth: { classPrefix: 'border-width', transformer: cssShorthandToClasses },
+    shadow: { classPrefix: 'shadow', transformer: generateResponsiveClasses },
+    fontSize: { classPrefix: 'font-size', transformer: generateResponsiveClasses },
+  };
+
+  const hoverClasses = hover
+    ? Object.entries(hover).map(([key, value]) => (
+      cssPropertyMap[key].transformer(`hover:${cssPropertyMap[key as keyof BoxProps['hover']].classPrefix}`, value)
+    ))
+    : undefined;
+
+  const focusClasses = focus
+    ? Object.entries(focus).map(([key, value]) => (
+      cssPropertyMap[key].transformer(`focus:${cssPropertyMap[key as keyof BoxProps['focus']].classPrefix}`, value)
+    ))
+    : undefined;
+
   const boxClasses = classNames(
+    styles.box,
```

Comment:
I think we should only apply the transition styles when a hover or focus is set.

Reply:
fari enough.